django_saml2_auth/views.py

```diff
@@ -117,11 +117,23 @@ def denied(r):
     return render(r, 'django_saml2_auth/denied.html')
 
 
-def _create_new_user(username, email, firstname, lastname):
-    user = user_model.objects.create_user(username, email)
-    user.first_name = firstname
-    user.last_name = lastname
-    groups = [Group.objects.get(name=x) for x in settings.SAML2_AUTH.get('NEW_USER_PROFILE', {}).get('USER_GROUPS', [])]
+def _create_new_user(user_identity):
+    attributes_map = settings.SAML2_AUTH.get('ATTRIBUTES_MAP', {})
+    user_name = user_identity[settings.SAML2_AUTH.get(
+        'ATTRIBUTES_MAP', {}).get('username', 'UserName')][0]
+    user = user_model.objects.create_user(user_name)
+    for key in attributes_map:
+        if key != 'username':
+            try:
+                if type(user_identity[attributes_map[key]]) is list:
+                    setattr(user, key, user_identity[attributes_map[key]][0])
+                else:
+                    setattr(user, key, user_identity[attributes_map[key]])
```

Comment:
Use `isinstance(..., list)` rather than `type(...) is list`.

This code would also be much cleaner if you created a variable for the value rather than repeatedly writing those long expressions many times, e.g.
```python
for (user_attr, saml_attr) in attributes_map.items():
    saml_value = user_identity[saml_attr]   # OR if we really like to hide the absense of the saml_attr, then this could use "user_identity.get(saml_attr)" followed by if "saml_value is None: continue", but I think that's bad idea
    value = saml_value[0] if isinstance(saml_value, list) else saml_value
    setattr(user, user_attr, value)
```